src/MudBlazor/Components/DataGrid/MudDataGrid.razor.cs

```
@@ -1280,6 +1255,34 @@ public override async Task SetParametersAsync(ParameterView parameters)
                 await ClearCurrentSortings();
         }
 
+        private void OnSelectedItemChanged(ParameterChangedEventArgs<T> args)
+        {
+            if (!MultiSelection)
+            {
+                Selection.Clear();
+            }
+
+            // add new item to Selection
+            if (!Selection.Remove(args.Value))
+            {
+                Selection.Add(args.Value);
+            }
+
+            _selectedItemsState.SetValueAsync(Selection);
```

This discards the task and thus may swallow exceptions from the user's code. There is a way to catch and log the exception from the task like this:

`_selectedItemsState.SetValueAsync(Selection).CatchAndLog();`

---

Why don't we want to make it async? The `void OnSelectedItemChanged` can be `Task OnSelectedItemChangedAsync`, parameter state supports both sync and async.

---

It should have been async Task, I just missed it was using a set async method.